dijkstra/lib/dijkstra.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const TinyQueue = require('tinyqueue');
+
+function shortestPath(graph, start, destination) {
+  const queue = new TinyQueue([], (a, b) => { return a.distance > b.distance; });
+  const distances = {};
+  const predecessor = {};
+
+  // initialize
+  distances[start] = 0;
+  queue.push({index: start, distance: distances[start]});
+
+  while (queue.length !== 0 && queue.peek().index !== destination) {
+    let node = queue.pop();
+
+    graph.nodes[node.index].edges.forEach((edge) => {
+      let tentativeWeight = distances[node.index] + edge.weight;
+      let distance = distances[edge.destination];
```

Comment:
Most of `let` statements can be converted `const` in the PR since they're not changed. 

Also,`const` and `let` are only about the reference — it doesn't matter if the object itself is muted. This is fine:

```js
const foo = [];
foo.push('bar');
```